### PR TITLE
3192 adds plot util

### LINKED ISSUE #3192: util to visualise 3d volumes
**Is your feature request related to a problem? Please describe.**
a utility method to generate figures for 3d volumes as a grid of 2d frames.

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #3192

### Description
Figure_1
Figure_2

### Status
**ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_matshow3d.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        )
+        self.assertTrue(mat.dtype == np.float32)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            tempimg = f"{tempdir}/matshow3d_patch_test.png"
+            fig.savefig(tempimg)
+            comp = compare_images(f"{testing_dir}/matshow3d_patch_test.png", tempimg, 5e-2, in_decorator=True)
+            if comp:
+                print("not none comp: ", comp)
+                np.testing.assert_allclose(comp["rms"], 29.11797096868759, atol=1e-3, rtol=1e-3)
```
Comment (wyli) [inline tests/test_matshow3d.py:70]: this is expected with matplotlib 3.2.2

File: monai/visualize/utils.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    vmax = np.nanmax(vol) if vmax is None else vmax
+
+    # making grid of every_n-th frame
+    vol = vol[:: max(every_n, 1)]
+    if not frames_per_row:
+        frames_per_row = int(np.ceil(np.sqrt(len(vol))))
+    frames_per_row = max(min(len(vol), frames_per_row), 1)
+    height, width = vol.shape[-2:]
+    n_rows = int(np.ceil(len(vol) / frames_per_row))
+    im = np.full(shape=(height * n_rows, width * frames_per_row), fill_value=fill_value, dtype=vol.dtype)
```
Comment (ericspod) [inline monai/visualize/utils.py:102]: I had in the past use something like the following to arrange a stack of images into a tile arrangement that might be a better solution:

```python
def tile_stack(stack,cols,rows=1):
    b=len(stack)
    stack=stack[:b-b%cols]
    rows=min(rows,len(stack)//cols)
    out=[]
    
    for r in range(rows):
        out.append([stack[r*cols+c] for c in range(cols)])
        
    return np.block(out)
```

Comment (wyli) [reply monai/visualize/utils.py]: Many thanks, the method is nice!

**Review by ericspod — APPROVED** [review]

**Comment by Nic-Ma** [comment]
Hi @ericspod and @wyli ,

Thanks for the great PR, please wait a while before merging it.
I will try to review it tonight.

Thanks.

File: requirements-dev.txt [hunk 3]
```diff
@@ -40,3 +40,4 @@ requests
 einops
 transformers
 mlflow
+matplotlib
```
Comment (Nic-Ma) [inline requirements-dev.txt:43]: Could you please help also add `matplotlib` to other config files for optional import packages?
Like `pip install monai[matplotlib]`.

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Put some minor comments inline.
Others look good to me.

Thanks.

**Pull request merged**